### PR TITLE
fix(snap): add ToC to snap/README.md

### PR DESCRIPTION
**Pull request by tonyespy**
This changes adds a manually generated ToC to the snap/README file. It also changes some of the heading capitalization to ensure style consistency.

### CODE REVIEW

**Comment by tonyespy** [comment]
@siggiskulason I've updated all the links and everything seems to be working OK now, so ready to review.

File: snap/README.md [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+  * [Using the EdgeX snap](#using-the-edgex-snap)
+    + [Configuring individual services](#configuring-individual-services)
+    + [Viewing logs](#viewing-logs)
+    + [Configuration overrides](#configuration-overrides)
+    + [Security services](#security-services)
+      - [Secret Store](#secret-store)
+      - [API Gateway](#api-gateway)
+        * [API Gateway user setup](#api-gateway-user-setup)
+          + [JWT tokens](#jwt-tokens)
+          + [OAuth 2.0](#oath-2.0)
```
Comment (siggiskulason) [inline snap/README.md:18]: This link doesn't work with the period in the version number - it needs to be "#oauth-20" (not #oauth-2.0)

**Review by siggiskulason — CHANGES_REQUESTED** [review]
Looks good - just one link that needs fixing - see inline comment

**Review by siggiskulason — APPROVED** [review]
LGTM

**Pull request merged**